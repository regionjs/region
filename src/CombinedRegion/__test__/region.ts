import { CombinedRegion } from '../..';

export const region = new CombinedRegion({ name: 'region', enableLog: true });
