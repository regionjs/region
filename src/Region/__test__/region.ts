import { Region } from '../..';

export const region = new Region({ name: 'region' });
