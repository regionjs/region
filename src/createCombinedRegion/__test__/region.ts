import { createCombinedRegion } from '../..';

export const region = createCombinedRegion();
