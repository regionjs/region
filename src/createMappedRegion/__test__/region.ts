import { createMappedRegion } from '../..';

export const region = createMappedRegion<string, any>();
