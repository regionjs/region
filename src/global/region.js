import Region from '../region/index';

export const region = new Region();
