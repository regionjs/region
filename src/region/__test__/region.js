import Region from '../index';
import { getProvider } from '../..';

getProvider();
const region = new Region({ name: null });

export { region };
