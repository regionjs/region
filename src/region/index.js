import wrapSetConfig from './wrapSetConfig';
import wrapReducer from './wrapReducer';
import wrapGet from './wrapGet';
import wrapPrivate from './wrapPrivate';
import wrapPublic from './wrapPublic';
import wrapReact from './wrapReact';

const compose = (...funcs) => funcs.reduce((a, b) => (...args) => a(b(...args)), arg => arg);

export default compose(
  wrapReact,
  wrapPublic,
  wrapPrivate,
  wrapGet,
  wrapReducer,
  wrapSetConfig,
)();
