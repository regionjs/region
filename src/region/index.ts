import wrapSetConfig from './wrapSetConfig';
import wrapReducer from './wrapReducer';
import wrapPrivate from './wrapPrivate';
import wrapPublic from './wrapPublic';
import wrapReact from './wrapReact';

/**
 * copy from recompose
 * @see https://github.com/acdlite/recompose/blob/master/src/packages/recompose/compose.js
 */
// @ts-ignore
const compose = (...funcs) => funcs.reduce((a, b) => (...args) => a(b(...args)), arg => arg);

export default compose(
  wrapReact,
  wrapPublic,
  wrapPrivate,
  wrapReducer,
  wrapSetConfig,
)();
