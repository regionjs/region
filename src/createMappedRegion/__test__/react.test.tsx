import * as React from 'react';
import * as reactTestRenderer from 'react-test-renderer';
import { region } from './region';

const { set, useLoading, useValue } = region;

const connect = (key: any) => (Component: any) => {
  return () => {
    const loading = useLoading(key);
    const value = useValue(key);
    const props = { [key]: value };
    return <Component loading={loading} {...props} />;
  };
};

const connectWith = (key: any, Component: any) => connect(key)(Component);

describe('react', () => {
  test('useProps', () => {
    const User: React.FC = () => {
      const loading = useLoading('user');
      const user = useValue('user');
      return loading ? <div>{`loading ${user}`}</div> : <div>{`!loading ${user}`}</div>;
    };
    expect(reactTestRenderer.create(<User />).toJSON()).toMatchSnapshot();
    set('user', 'user');
    expect(reactTestRenderer.create(<User />).toJSON()).toMatchSnapshot();
  });

  test('connect and connectWith', () => {
    const User: React.FC = ({ loading, user }: any) => loading ? <div>{`loading ${user}`}</div> : <div>{`!loading ${user}`}</div>;
    const ConnectUser = connect('user')(User);
    expect(reactTestRenderer.create(<ConnectUser />).toJSON()).toMatchSnapshot();
    set('user', 'user2');
    expect(reactTestRenderer.create(<ConnectUser />).toJSON()).toMatchSnapshot();
    const ConnectWithUser = connectWith('user', User);
    expect(reactTestRenderer.create(<ConnectWithUser />).toJSON()).toMatchSnapshot();
    set('user', 'user3');
    expect(reactTestRenderer.create(<ConnectWithUser />).toJSON()).toMatchSnapshot();
  });
});
