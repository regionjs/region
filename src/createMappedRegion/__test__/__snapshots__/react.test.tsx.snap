// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`react > connect and connectWith 1`] = `
<div>
  !loading user
</div>
`;

exports[`react > connect and connectWith 2`] = `
<div>
  !loading user2
</div>
`;

exports[`react > connect and connectWith 3`] = `
<div>
  !loading user2
</div>
`;

exports[`react > connect and connectWith 4`] = `
<div>
  !loading user3
</div>
`;

exports[`react > useProps 1`] = `
<div>
  loading undefined
</div>
`;

exports[`react > useProps 2`] = `
<div>
  !loading user
</div>
`;
